### PR TITLE
[WIP] Th apex

### PR DESCRIPTION
**Pull request by taylorharwin**
- Initial rough start to add Apex processor token support to the ruby library.

### CODE REVIEW

File: lib/plaid/client.rb [hunk 1]
```diff
@@ -96,9 +96,9 @@ def transactions
     # public_key - The String Plaid account public key to authenticate requests
     def initialize(env:, client_id:, secret:, public_key:)
       @env        = env_map(env)
-      @client_id  = client_id
-      @secret     = secret
-      @public_key = public_key
+      @client_id  = '579947870259902a3980f4aa'
+      @secret     = '636d086312bec8166dc61ce25e16f8'
+      @public_key = '5f7b2a92afa8cfc8e5869050709fd2'
```
Comment (edmundloo) [inline lib/plaid/client.rb]: This isn't good! Make sure to remove these changes and change your secret!

File: lib/plaid/products/apex.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                             payload)
+    end
+  end
+
+  # Public: Class used to call the Apex sub-product.
+  class Apex
+    # Public: Memoized class instance
+    # to make requests from Plaid::ProcessorToken
+    def processor_token
+      @bank_account_token ||= Plaid::ProcessorToken.new(@client)
```
Comment (edmundloo) [inline lib/plaid/products/apex.rb]: This should be @processor_token here.

File: test/test_apex.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    @access_token = @item['access_token']
+  end
+
+  def teardown
+    @client.item.delete(@access_token)
+  end
+
+  def test_apex_bank_account_token_create_invalid_account_id
+    assert_raises(Plaid::InvalidRequestError) do
+      @client.processor.apex.processor_token.create(@access_token, BAD_STRING)
```
Comment (edmundloo) [inline test/test_apex.rb]: The path for this is wrong, the test should be failing for this case.

File: lib/plaid/products/apex.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    # be used to generate a Apex bank account token for a given account ID.
+    #
+    # access_token - access_token to create a public token for
+    # account_id - ID of the account to create a bank account token for
+    #
+    # Returns a parsed JSON containing a Apex bank account token
+    def create(access_token, account_id)
+      payload = { access_token: access_token,
+                  account_id: account_id }
+      @client.post_with_auth('processor/apex/processor_token/create ',
```
Comment (edmundloo) [inline lib/plaid/products/apex.rb]: Is the space as the end of the path intentional? Also, is the path supposed to be prefixed with "processor"? If yes, then the logic for this belongs under our processor file: https://github.com/plaid/plaid-ruby/blob/master/lib/plaid/products/processor.rb

File: lib/plaid/products/apex.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+module Plaid
+  # Public: Class used to call the Processor Token sub-product.
```
Comment (edmundloo) [inline lib/plaid/products/apex.rb]: nit: Space not supposed to be between Processor and Token.

File: lib/plaid/products/apex.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                  account_id: account_id }
+      @client.post_with_auth('processor/apex/processor_token/create ',
+                             payload)
+    end
+  end
+
+  # Public: Class used to call the Apex sub-product.
+  class Apex
+    # Public: Memoized class instance
+    # to make requests from Plaid::ProcessorToken
```
Comment (edmundloo) [inline lib/plaid/products/apex.rb]: I think you can fit this all on 1 line.

File: lib/plaid/products/apex.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      @bank_account_token ||= Plaid::ProcessorToken.new(@client)
+    end
+
+    def initialize(client)
+      @client = client
+    end
+  end
+
+  # Public: Class used to call the Processor product.
+  class Processor
```
Comment (edmundloo) [inline lib/plaid/products/apex.rb]: We don't want a duplicate Processor class if we can put them under a single one: https://github.com/plaid/plaid-ruby/blob/master/lib/plaid/products/processor.rb

**Comment by be9** [comment]
Closing in favor of #176.

**Pull request closed without merging**